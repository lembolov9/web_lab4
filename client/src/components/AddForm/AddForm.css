.AddForm {
    margin: 0 20px;
    display: flex;
    justify-content: space-between;
}

.addBtn {
    border-radius: 50%;
}
