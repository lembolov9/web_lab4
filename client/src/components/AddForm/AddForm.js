import React from "react";
import {connect} from "react-redux";
import {getCities} from "../../reducers";
import {addCity, addCityError, addCitySuccess} from "../../actions";
import "./AddForm.css"
import {api} from "../../constants/server";
import axios from "axios";

const addCityAction = name => {
    return dispatch => {
        dispatch(addCity(name));
        axios
            .post(`${api}favourites`, { name })
            .then(response => {
                const city = {
                    id: response.data.city._id,
                    name: response.data.city.name
                };
                dispatch(addCitySuccess(city));
            })
            .catch(error => {
                dispatch(addCityError(error));
            });
    };
};

class AddForm extends React.Component {
    constructor(props) {
        super(props);
        this.handleFormInput = this.handleFormInput.bind(this);
    }

    handleFormInput(event) {
        event.preventDefault();
        this.props.addCity(event.target[0].value);
    }

    render() {
        return (
            <div className="AddForm">
                <div><h2>Favourites</h2></div>
                <div>
                    <form onSubmit={this.handleFormInput}>
                        <input id="addCity" placeholder="Add city" required/>
                        <button className="addBtn" type="submit">+</button>
                    </form>
                </div>
            </div>
        );
    }
}

const mapStateToProps = state => ({
    cities: getCities(state)
});

const mapDispatchToProps = dispatch => ({
    addCity: name => dispatch(addCityAction(name))
});

const Add = connect(mapStateToProps, mapDispatchToProps)(AddForm);


export {Add, AddForm};