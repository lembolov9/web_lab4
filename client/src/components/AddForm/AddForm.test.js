import renderer from "react-test-renderer";
import React from "react";
import {AddForm} from "./AddForm";


it("AddForm render ok", () => {
    const component = renderer.create(<AddForm />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});