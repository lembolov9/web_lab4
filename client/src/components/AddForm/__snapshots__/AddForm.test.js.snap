// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddForm render ok 1`] = `
<div
  className="AddForm"
>
  <div>
    <h2>
      Favourites
    </h2>
  </div>
  <div>
    <form
      onSubmit={[Function]}
    >
      <input
        id="addCity"
        placeholder="Add city"
        required={true}
      />
      <button
        className="addBtn"
        type="submit"
      >
        +
      </button>
    </form>
  </div>
</div>
`;
